llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -2337,10 +2340,11 @@ class BoUpSLP {
       assert((empty() || VL.size() == getNumLanes()) &&
              "Expected same number of lanes");
       assert(isa<Instruction>(VL[0]) && "Expected instruction");
+      unsigned NumOperands = cast<Instruction>(VL[0])->getNumOperands();
+      // IntrinsicInst::isCommutative returns true if swapping the first "two"
+      // arguments to the intrinsic produces the same result.
       constexpr unsigned IntrinsicNumOperands = 2;
-      unsigned NumOperands = isa<IntrinsicInst>(VL[0])
-                                 ? IntrinsicNumOperands
-                                 : cast<Instruction>(VL[0])->getNumOperands();
+      Arg_size = isa<IntrinsicInst>(VL[0]) ? IntrinsicNumOperands : NumOperands;
```

Comment:
Still do not understand why we need Arg_size here. BTW, Better it use NumOperands or ArgSize name

Reply:
We need `ArgSize` to make sure `reorder` only swap the first two operands. After this PR, `OpsVec` also contains non swappable operands (because we want `appendOperandsOfVL` can be reused even the operands of VL is not commutative).